src/vanilla.cpp

```diff
@@ -67,19 +67,18 @@ Rcpp::List europeanOptionEngine(std::string type,
     boost::shared_ptr<QuantLib::StrikedTypePayoff> payoff(new QuantLib::PlainVanillaPayoff(optionType, strike));
     
     if (withDividends) {
-        Rcpp::NumericVector divvalues(discreteDividends), divtimes(discreteDividendsTimeUntil);
-        int n = divvalues.size();
+        std::vector<double> discDividends = Rcpp::as<std::vector<double> >(discreteDividends);
+        std::vector<double> divtimes = Rcpp::as<std::vector<double> >(discreteDividendsTimeUntil);
+        int n = divtimes.size();
         std::vector<QuantLib::Date> discDivDates(n);
-        std::vector<double> discDividends(n);
-        boost::posix_time::time_duration discreteDividendLength;        
         for (int i = 0; i < n; i++) {
 #ifdef QL_HIGH_RESOLUTION_DATE
-            discreteDividendLength = boost::posix_time::minutes(divtimes[i] * 360 * 24 * 60);
+            boost::posix_time::time_duration discreteDividendLength =
```

Comment:
No it didn't and couldn't. Do you even read my comment about boost::posix_time::time_duration not being defined? There is a reason why I move it inside the guard, it actually makes it compile if  QL_HIGH_RESOLUTION_DATE=0!